x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/cache/CacheFile.java

```diff
@@ -494,25 +496,30 @@ private void markAsNeedsFSync() {
                         // Capture the completed ranges before fsyncing; ranges that are completed after this point won't be considered as
                         // persisted on disk by the caller of this method, even if they are fully written to disk at the time the file
                         // fsync is effectively executed
-                        final SortedSet<Tuple<Long, Long>> completedRanges = tracker.getCompletedRanges();
-                        assert completedRanges != null;
-                        assert completedRanges.isEmpty() == false;
+                        final SortedSet<Tuple<Long, Long>> fsyncedRanges = tracker.getCompletedRanges();
+                        assert fsyncedRanges != null;
+                        assert fsyncedRanges.isEmpty() == false;
 
-                        IOUtils.fsync(file, false, false);
+                        if (evicted.get() == false) {
```

Comment:
Checking the eviction here does not solve the issue described in this pull request. It is more another best effort to not fsync files that we know are evicted.